source/nodejs/adaptivecards-designer/src/components/treeview.ts

```diff
@@ -0,0 +1,132 @@
+import * as Adaptive from "adaptivecards";
+import * as Designer from "../card-designer";
+import * as Utils from "../utils"
+import { AdaptiveCard } from "adaptivecards";
+
+export default class Treeview {
```

Comment:
Please consider reworking this [according to my treeview implementation](https://github.com/Microsoft/AdaptiveCards/pull/1613/commits/fba89e960809ecb8f1c8918a729236a74fc213e7) or simply replacing with my implementation. The treeview should be entirely based on the existing peer hierarchy. Peers are the core of the designer, and implement important behaviors, such as element removal, selection, drop, and more. For instance, I can imagine in a later version of the designer that a right click on a treeview item would drop down a menu listing the peer's commands, allowing the user to, say, add a column right from within the tree view.

Reply:
Peers should have a new `typeRegistration` property that points back to the ITypeRegistration that they were created from. This way, you can also retrieve the peer's iconUrl, which you need to build the tree.

Reply:
In progress.